Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/Function.h

```diff
@@ -11,15 +11,20 @@
 #include <AK/RefPtr.h>
 #include <AK/StringView.h>
 
+#include "DiagnosticEngine.h"
 #include "Forward.h"
 
 namespace JSSpecCompiler {
 
 struct TranslationUnit {
+    TranslationUnit(StringView filename);
+    ~TranslationUnit();
+
     void adopt_declaration(NonnullRefPtr<FunctionDeclaration>&& declaration);
     FunctionDefinitionRef adopt_function(NonnullRefPtr<FunctionDefinition>&& definition);
 
     StringView filename;
+    DiagnosticEngine diagnostic_engine;
```

Comment:
If the translation unit owns the Diagnostic engine, why does it need to be globally accessible? Can't the parser just do translation_unit->diag().fatal_error() or some such?